pkg/gen/typegen.go

```diff
@@ -918,3 +924,52 @@ func additionalSecretOutputs(gvk schema.GroupVersionKind) []string {
 		return []string{}
 	}
 }
+
+func aliasesForGVK(gvk schema.GroupVersionKind) []string {
+	kind := kinds.Kind(gvk.Kind)
+
+	// It's unsafe to move between `extensions/v1beta1`, and the newer apiVersions due to differences in
+	// the behavior of the Deployment and ReplicaSet. Even if the apiVersion is changed,
+	// the resource will continue to use the old behavior, which will break the await logic. Without an
+	// alias set, the engine will recreate the resource with the newer apiVersion.
+	if gvk.GroupVersion().String() == "extensions/v1beta1" {
+		switch kind {
+		case kinds.DaemonSet, kinds.Deployment, kinds.ReplicaSet, kinds.StatefulSet:
+			return []string{}
+		}
+	}
+
+	switch kind {
+	case kinds.DaemonSet:
+		return []string{
+			"kubernetes:apps/v1:DaemonSet",
+			// For some reason, there is no `apps/v1beta1:DaemonSet`.
```

Comment:
There is, it's under `extensions`: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.16/#daemonset-v1beta1-extensions